chainer/backends/cuda.py

```diff
@@ -168,8 +168,16 @@ def get_device_from_array(*arrays):
 
     The device on which the given CuPy array reside is returned.
 
+    .. note::
+
+        This method only recognizes :class:`cupy.ndarray`\\ s in arguments.
+        Especially note that, unlike :func:`get_array_module`, this method
+        does not recognize :class:`~chainer.Variable` objects.
+        If you need to get device from the :class:`~chainer.Variable` instance
+        ``v``, you need to use ``get_device_from_array(v.data)``.
```

Comment:
Accessing the array object via `data` attribute is discouraged. (Please see: [Variable.data](https://docs.chainer.org/en/stable/reference/core/generated/chainer.Variable.html?highlight=Variable#chainer.Variable.data))
So, I think it's better to use `v.array` here.

Reply:
Thank you, fixed.